test/FastTests/Issues/RavenDB_13891.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Commands.Batches;
+using Raven.Client.Documents.Operations;
+using Xunit;
+
+namespace FastTests.Issues
+{
+    public class RavenDB_13891 : RavenTestBase
+    {
+        [Fact]
+        public async Task PatchedDocumentShouldBeReplaced()
+        {
+            using (var store = GetDocumentStore())
+            {
+                string personId = null;
+                using (var s = store.OpenAsyncSession())
+                {
+                    for (var i = 1; i < 3; i++)
+                    {
+                        var p = new Person
+                        {
+                            Modifications = new List<Person.UserChange>
+                            {
+                                new Person.UserChange
+                                {
+                                    Id = $"{i}",
+                                    Timestamp = DateTime.Now
+                                }
+                            }
+                        };
+                        await s.StoreAsync(p);
+                        personId = p.Id;
+                    }
+
+                    await s.SaveChangesAsync();
+
+                    s.Advanced.Defer(
+                        new PatchCommandData(
+                            personId,
+                            null,
+                            new PatchRequest
+                            {
+                                Script = $@"this.Text = 'aaa';"
+                            },
+                            null));
+
+                    var person = await s.LoadAsync<Person>(personId);
+                    Assert.True(string.IsNullOrEmpty(person.Text));
+                    Assert.Equal(1, person.Modifications.Count);
+                    await s.SaveChangesAsync();
+
+                    person = await s.LoadAsync<Person>(personId);
+                    Assert.Equal("aaa", person.Text);
+                    Assert.Equal(1, person.Modifications.Count);
+                }
+            }
+        }
+
+        public class Person
```

Comment:
private

Reply:
What is the reasoning here? This is inside a test class.

Reply:
Good practice. E.g. to avoid using it from different tests (happens often). Tests should be isolated and use only their own types or common ones defined elsewhere if possible.

Reply:
done